applications/fishing-map/src/features/feedback/FeedbackModal.tsx

```diff
@@ -0,0 +1,207 @@
+  const [suficientData, setSuficientData] = useState(false)
+  const [feedbackData, setFeedbackData] = useState<FeedbackData>({
+    url: window.location.href,
+    date: new Date().toString(),
+  })
+  const activeDataviews = useSelector(selectActiveDataviews)
+  const userData = useSelector(selectUserData)
+  const datasetOptions = activeDataviews.flatMap((dataview) => {
+    if (dataview.config?.type === Generators.Type.HeatmapAnimated) {
+      return getSourcesSelectedInDataview(dataview)
+    } else {
+      return {
+        id: dataview.id,
+        label: t(`datasets:${dataview.datasets?.[0].id.split(':')[0]}.name` as any),
+      }
+    }
+  })
+
+  const onNameChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
+    setFeedbackData({ ...feedbackData, name: target.value })
+  }
+
+  const onEmailChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
+    setFeedbackData({ ...feedbackData, email: target.value })
+  }
+
+  useEffect(() => {
+    const name = userData ? `${userData.firstName} ${userData.lastName || ''}` : ''
+    setFeedbackData({
+      ...feedbackData,
+      userId: userData?.id || GUEST_USER_TYPE,
+      email: userData?.email || '',
+      name: name,
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userData])
+
+  const onInstitutionChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
+    setFeedbackData({ ...feedbackData, institution: target.value })
+  }
+
+  const onSelectRole = (option: SelectOption) => {
+    setFeedbackData({ ...feedbackData, role: option.label })
+  }
+
+  const onRemoveRole = (option: SelectOption) => {
+    setFeedbackData({ ...feedbackData, role: '' })
+  }
+
+  const onSelectDataset = (option: SelectOption) => {
+    setFeedbackData({ ...feedbackData, dataset: option.label })
+  }
+
+  const onRemoveDataset = (option: SelectOption) => {
+    setFeedbackData({ ...feedbackData, dataset: '' })
+  }
+
+  // const [description, setDescription] = useState('')
+  const onDescriptionChange = ({ target }: ChangeEvent<HTMLTextAreaElement>) => {
+    setFeedbackData({ ...feedbackData, description: target.value })
+  }
+
+  useEffect(() => {
+    const { description, name, email } = feedbackData
+    const hasMandatoryData =
+      description !== undefined &&
+      description !== '' &&
+      name !== undefined &&
+      name !== '' &&
+      validateEmail(email || '')
+    setSuficientData(hasMandatoryData)
+  }, [feedbackData, userData])
+
+  const sendFeedback = async () => {
+    try {
+      setLoading(true)
+      await doc.useServiceAccountAuth({
+        client_email: FEEDBACK_CLIENT_EMAIL,
+        private_key: FEEDBACK_PRIVATE_KEY,
+      })
+      // loads document properties and worksheets
+      await doc.loadInfo()
+      const sheet = doc.sheetsById[FEEDBACK_SHEET_ID]
+      await sheet.addRow(feedbackData)
+      setLoading(false)
+      setFeedbackData({})
+      onClose()
+    } catch (e) {
+      console.error('Error: ', e)
```

Comment:
When failing the loading is going to keep as it is forever, I'd handle the error with some feedback to the user